kernel/thread.c

```diff
@@ -286,6 +286,10 @@ void _setup_new_thread(struct k_thread *new_thread,
 
 	/* Any given thread has access to itself */
 	k_object_access_grant(new_thread, new_thread);
+
+	if (options & K_INHERIT_PERMS) {
```

Comment:
I have a doubt here:
K_INHERIT_PERMS option can be used by kernel threads as well but is enclosed on CONFIG_USERSPACE, what is intended here? we are allowing permission inheritance only if USERSPACE feature is enabled? or it is only intended for user threads?

Reply:
there's no notion of permissions outside the CONFIG_USERSPACE feature as they only are applicable to user threads. Without userspace, all threads are supervisor and can do whatever they want.

Reply:
we want people to be able to switch CONFIG_USERSPACE on and off without changing their code so some of the APIs/defines are no-ops if the feature is disabled.

Reply:
ok got it. Thanks.